events/utils.py

```diff
@@ -32,8 +32,9 @@ def get_value_from_tuple_list(list_of_tuples, search_key, value_index):
                         returned and which is used as a key
     :return: Value from either side of tuple
     """
+
     for i, v in enumerate(list_of_tuples):
-        if v[value_index ^ 1] == search_key:
+        if str(v[value_index ^ 1]) == str(search_key):
```

Comment:
I don't get the idea of this `^1`. What it does? 🤔

Reply:
this is a bitwise comparison. This function takes in a list of tuples and then treats it as a dictionary, with either the first or the second entry of the tuple being the key, and the other, respectively, the value.
Let's call this whole function g(list_of_tuples, search_key, index) and let's say we have a list of tuples  l = [('a',1),('b',2),('c',3)], then 
g(l, 'a', 0) is None
g(l, 'a', 1) == 1
g(l, 1, 0) == 'a'
g(l, 1, 1) is None